datafusion/core/tests/parquet/filter_pushdown.rs

```diff
@@ -638,9 +638,28 @@ async fn predicate_cache_pushdown_default() -> datafusion_common::Result<()> {
     // The cache is on by default, and used when filter pushdown is enabled
     PredicateCacheTest {
         expected_inner_records: 8,
-        // reads more than necessary from the cache as then another bitmap is applied
-        // See https://github.com/apache/datafusion/pull/18820 for setting and workaround
-        expected_records: 7,
+        expected_records: 7, // reads more than necessary from the cache as then another bitmap is applied
```

Comment:
this test shows the change in behavior after the parquet 57.1.0 upgrade. The previous result with 57.0.0 was `4`

The old result can now be obtained by setting `force_filter_selections` to true